engine/apps/alerts/models/alert_group.py

```diff
@@ -420,8 +422,17 @@ def status(self) -> int:
     # https://code.djangoproject.com/ticket/28545
     is_open_for_grouping = models.BooleanField(default=None, null=True, blank=True)
 
+    # todo: rework using this field to use DeclaredIncident model field instead
     grafana_incident_id = models.CharField(max_length=100, null=True, default=None)
 
+    declared_incident = models.ForeignKey(
+        "alerts.DeclaredIncident",
+        on_delete=models.SET_NULL,
+        null=True,
+        default=None,
+        related_name="attached_alert_groups",
+    )
```

Comment:
Mainly as a way to simplify/enforce checks (like having one-active incident per route, or a potential limit of created incidents per org per timeframe) while reducing the incident API calls (and avoiding "complex/big" queries to fetch the information needed; this way we only create incident or query for an incident using the ID).
Eventually we could also store additional metadata in the relation (and as commented in the code, it may make sense to migrate the existing field to use the relation instead).

Reply:
makes sense! was just curious, thanks for clarifying!